### ISSUE TITLE
Azure Gateway Lifecyle support

### ISSUE THREAD

**Issue by jonesbusy** [minio/minio#8097]
Hi,

**Is your feature request related to a problem? Please describe.**

Bucket lifecycle was implemented on 'RELEASE.2019-08-14T20-37-41Z' but it's not possible to use it with Minio acting as Azure Gateway.

Azure Blob storage fully support lifecycle : https://docs.microsoft.com/en-us/azure/storage/blobs/storage-lifecycle-management-concepts

But we cannot use it with Minio to have a S3 compatible API.

**Describe the solution you'd like**

Bucket Lifecycle management with Azure Gateway

**Additional context**

To reproduce

```
docker run -e MINIO_ACCESS_KEY=<access key of blob storage> -e MINIO_SECRET_KEY=<secret key of blob storage> -p 9000:9000 minio/minio:RELEASE.2019-08-14T20-37-41Z gateway azure
```

```
aws --debug s3api put-bucket-lifecycle --bucket temporary --endpoint-url http://localhost:9000 --lifecycle-configuration file://bucket-lifecycle.json
```

```
2019-08-17 14:53:11,014 - MainThread - botocore.hooks - DEBUG - Event after-call.s3.PutBucketLifecycle: calling handler <function enhance_error_msg at 0x7f256b623378>
2019-08-17 14:53:11,017 - MainThread - awscli.clidriver - DEBUG - Exception caught in main()
Traceback (most recent call last):
  File "/home/adelaye/.local/lib/python3.6/site-packages/awscli/clidriver.py", line 217, in main
    return command_table[parsed_args.command](remaining, parsed_args)
  File "/home/adelaye/.local/lib/python3.6/site-packages/awscli/clidriver.py", line 358, in __call__
    return command_table[parsed_args.operation](remaining, parsed_globals)
  File "/home/adelaye/.local/lib/python3.6/site-packages/awscli/clidriver.py", line 530, in __call__
    call_parameters, parsed_globals)
  File "/home/adelaye/.local/lib/python3.6/site-packages/awscli/clidriver.py", line 650, in invoke
    client, operation_name, parameters, parsed_globals)
  File "/home/adelaye/.local/lib/python3.6/site-packages/awscli/clidriver.py", line 662, in _make_client_call
    **parameters)
  File "/home/adelaye/.local/lib/python3.6/site-packages/botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/adelaye/.local/lib/python3.6/site-packages/botocore/client.py", line 661, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (NotImplemented) when calling the PutBucketLifecycle operation: A header you provided implies functionality that is not implemented
2019-08-17 14:53:11,018 - MainThread - awscli.clidriver - DEBUG - Exiting with rc 255

An error occurred (NotImplemented) when calling the PutBucketLifecycle operation: A header you provided implies functionality that is not implemented
```

Thanks again for the great work!

**Comment by nitisht** [comment]
@jonesbusy we'll discuss internally if there are plans to extend lifecycle support to gateways

**Comment by harshavardhana** [comment]
Closing this gateway issue to be done as a future milestone.

**Comment by fl-max** [comment]
@harshavardhana Was this support added? Is there another feature issue being used to track this?

This should fix [#7200](https://github.com/minio/minio/issues/7200)

**Comment by harshavardhana** [comment]
> @harshavardhana Was this support added? Is there another feature issue being used to track this?
> 
> This should fix [#7200](https://github.com/minio/minio/issues/7200)

No we have decided to freeze gateway features, there will be no more features added unless a future requirement comes in.
